### PR TITLE
Move recent react commits

### PR DESCRIPTION
**Pull request by asudoh**
Moves recent commits that haven't been merged in `carbon-components-react` to here. The commit history gets my author which is not 100% clean. Anybody feel free to suggest an alternate way.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature-request-or-enhancement---.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
```
Comment (emyarod) [inline .github/ISSUE_TEMPLATE/feature-request-or-enhancement---.md]: this change was already moved over I believe https://github.com/carbon-design-system/carbon/blob/master/.github/ISSUE_TEMPLATE/feature-request-or-enhancement.md

**Review by emyarod — APPROVED** [review]
looks good to me, just one note about the duplicate issue template and the `Tabs-story` conflict

File: .github/ISSUE_TEMPLATE/feature-request-or-enhancement---.md (hunk 1 above)
Comment (asudoh) [reply .github/ISSUE_TEMPLATE/feature-request-or-enhancement---.md]: Good point, I think recent merge in this PR subsumed this.

**Review by dakahn — APPROVED** [review]

**Comment by joshblack** [comment]
Seems like the tab test might be messed up?

**Comment by joshblack** [comment]
@asudoh RE: the author note, would you want to use `git subtree` to pull the commits over with the original author? Can then do a merge commit and keep them all without squashing

**Comment by asudoh** [comment]
@joshblack Thank you for the suggestion, tried that, but merge-to-master seems to have ended up with moving the clock backward instead of forward. Need more investigation. Meanwhile I'll close this PR to avoid accidental merge.

**Pull request closed without merging**